File: FastMines_swing/src/main/java/fmg/swing/app/serializers/UserSerializer.java
package fmg.swing.app.serializers;

import java.io.IOException;
import java.io.ObjectInput;
import java.io.ObjectOutput;
import java.util.UUID;

import fmg.core.app.ISerializer;
import fmg.core.types.model.User;

public class UserSerializer implements ISerializer {

    private static final long VERSION = 1;

    public void write(User user, ObjectOutput out) throws IOException {
        out.writeLong(VERSION);
        out.writeUTF(user.getGuid().toString());
        out.writeUTF(user.getName());

        String pass = user.getPassword();
        boolean exist = (pass != null) && !pass.isEmpty();
        out.writeBoolean(exist);
        if (exist)
            out.writeUTF(pass);

        exist = (user.imgAvatar != null) && !user.imgAvatar.isEmpty();
        out.writeBoolean(exist);
        if (exist)
            out.writeUTF(user.imgAvatar);
    }

    public User read(ObjectInput in) throws IOException {
        long ver = in.readLong();
        if (ver != VERSION)
            throw new IllegalArgumentException(UserSerializer.class.getSimpleName() + ": Unsupported version #" + ver);

        User user = new User();
        user.setGuid(UUID.fromString(in.readUTF()));
        user.setName(in.readUTF());

        boolean exist = in.readBoolean();
        if (exist)
            user.setPassword(in.readUTF());

        exist = in.readBoolean();
        if (exist)
            user.imgAvatar = in.readUTF();

        return user;
    }

}

File: FastMines_core/src/fmg/core/app/serializers/UserSerializer.cs
using System;
using System.IO;
using Fmg.Core.App.Model;

namespace Fmg.Core.App.Serializers {

    /// <summary> User (de)serializer </summary>
    public class UserSerializer : ISerializer {

        private const long VERSION = 1;

        public void Write(User user, BinaryWriter output) {
            output.Write(VERSION);
            output.Write(user.Id.ToString());
            output.Write(user.Name);

            string pass = user.Password;
            bool exist = string.IsNullOrEmpty(pass);
            output.Write(exist);
            if (exist)
                output.Write(pass);

            exist = string.IsNullOrEmpty(user.ImgAvatar);
            output.Write(exist);
            if (exist)
                output.Write(user.ImgAvatar);
        }

        public User Read(BinaryReader input) {
            long ver = input.ReadInt64();
            if (ver != VERSION)
                throw new ArgumentException(nameof(UserSerializer) + ": Unsupported version #" + ver);

            User user = new User {
                Id = new Guid(input.ReadString()),
                Name = input.ReadString()
            };

            bool exist = input.ReadBoolean();
            if (exist)
                user.Password = input.ReadString();

            exist = input.ReadBoolean();
            if (exist)
                user.ImgAvatar = input.ReadString();

            return user;
        }

}

}
